### ISSUE TITLE
Upgrade from 2.5.2 to 2.5.3 Service Manager Woes

### ISSUE THREAD

**Issue by wshafer** [zendframework/zf2#7671]
Hey Guys,

After an update tonight I got ZF 2.5.3 (was 2.5.2)...  Immediately after I started to get the error:

```
Warning: array_key_exists(): The first argument should be either a string or an integer in /vagrant/src/cms/vendor/zendframework/zend-eventmanager/src/SharedEventManager.php on line 153
```

There's clearly a BC break in here but I can't find any documentation anywhere on this guy.

I've had to revert back to 5.3.2 for the time being.

Related: [https://github.com/zendframework/zend-cache/issues/71]

**Comment by wshafer** [comment]
FYI, It seems, that with @weierophinney last commit, all the packages are updating to 2.6, 2.7 ect.  I'm not sure if this was intended behavior or not.

**Comment by Ocramius** [comment]
Could you check the specific package changes? The main repo for 2.5.x is empty, so please diff your `composer.lock`

**Comment by wshafer** [comment]
```
Loading composer repositories with package information
Updating dependencies (including require-dev)
  - Removing zendframework/zend-stdlib (2.5.2)         
  - Installing zendframework/zend-stdlib (2.7.4)
    Loading from cache

  - Installing zendframework/zend-hydrator (1.0.0)
    Loading from cache

  - Removing zendframework/zend-eventmanager (2.5.2)
  - Installing zendframework/zend-eventmanager (2.6.2)
    Loading from cache

  - Removing zendframework/zend-code (2.5.3)
  - Installing zendframework/zend-code (2.6.2)
    Loading from cache

  - Removing zendframework/zend-server (2.5.1)
  - Installing zendframework/zend-server (2.6.1)
    Loading from cache

  - Removing zendframework/zend-math (2.5.2)
  - Installing zendframework/zend-math (2.6.0)
    Loading from cache

  - Removing zendframework/zend-json (2.5.2)
  - Installing zendframework/zend-json (2.6.1)
    Loading from cache

  - Installing container-interop/container-interop (1.1.0)
    Loading from cache

  - Removing zendframework/zend-servicemanager (2.5.1)
  - Installing zendframework/zend-servicemanager (2.7.5)
    Loading from cache

  - Removing zendframework/zend-text (2.5.1)
  - Installing zendframework/zend-text (2.6.0)
    Loading from cache

  - Removing zendframework/zend-filter (2.5.1)
  - Installing zendframework/zend-filter (2.6.1)
    Loading from cache

  - Removing zendframework/zend-form (2.5.3)
  - Installing zendframework/zend-form (2.6.0)
    Loading from cache

  - Removing zendframework/zend-mvc (2.5.3)
  - Installing zendframework/zend-mvc (2.6.0)
    Loading from cache

  - Removing zendframework/zend-dom (2.5.1)
  - Installing zendframework/zend-dom (2.6.0)
    Loading from cache

  - Removing zendframework/zend-console (2.5.1)
  - Installing zendframework/zend-console (2.6.0)
    Loading from cache

  - Removing zendframework/zend-tag (2.5.1)
  - Installing zendframework/zend-tag (2.6.1)
    Loading from cache

  - Removing zendframework/zend-serializer (2.5.1)
  - Installing zendframework/zend-serializer (2.6.1)
    Loading from cache

  - Removing zendframework/zend-permissions-acl (2.5.1)
  - Installing zendframework/zend-permissions-acl (2.6.0)
    Loading from cache

  - Removing zendframework/zend-modulemanager (2.5.3)
  - Installing zendframework/zend-modulemanager (2.6.1)
    Loading from cache

  - Removing zendframework/zend-crypt (2.5.2)
  - Installing zendframework/zend-crypt (2.6.0)
    Loading from cache

  - Installing psr/log (1.0.0)
    Loading from cache

  - Removing zendframework/zend-log (2.5.2)
  - Installing zendframework/zend-log (2.7.0)
    Loading from cache

  - Removing zendframework/zend-i18n (2.5.1)
  - Installing zendframework/zend-i18n (2.6.0)
    Loading from cache

  - Removing zendframework/zend-feed (2.5.2)
  - Installing zendframework/zend-feed (2.7.0)
    Loading from cache

  - Removing zendframework/zend-db (2.5.2)
  - Installing zendframework/zend-db (2.6.2)
    Loading from cache

  - Removing zendframework/zend-config (2.5.1)
  - Installing zendframework/zend-config (2.6.0)
    Loading from cache

  - Removing zendframework/zend-cache (2.5.3)
  - Installing zendframework/zend-cache (2.6.0)
    Loading from cache

  - Removing zendframework/zendframework (2.5.2)
  - Installing zendframework/zendframework (2.5.3)
    Loading from cache
```

**Comment by Ocramius** [comment]
Yeah, that's a huge jump/upgrade. More details about the trace that caused
the problem are needed...

**Comment by wshafer** [comment]
Here's one of the errors...  There's TONS being drawn up and of course eventually the browser crashes.

```
Warning: array_key_exists(): The first argument should be either a string or an integer in /vagrant/src/cms/vendor/zendframework/zend-eventmanager/src/SharedEventManager.php on line 153

Call Stack:
    0.0002     233792   1. {main}() /vagrant/src/cms/public/index.php:0
    0.0055     607984   2. Zend\Mvc\Application::init() /vagrant/src/cms/public/index.php:28
    0.0988    2701792   3. Zend\Mvc\Application->bootstrap() /vagrant/src/cms/vendor/zendframework/zend-mvc/src/Application.php:261
    0.1141    3079920   4. Zend\EventManager\EventManager->trigger() /vagrant/src/cms/vendor/zendframework/zend-mvc/src/Application.php:157
    0.1141    3080216   5. Zend\EventManager\EventManager->triggerListeners() /vagrant/src/cms/vendor/zendframework/zend-eventmanager/src/EventManager.php:214
    0.1441    3622672   6. call_user_func:{/vagrant/src/cms/vendor/zendframework/zend-eventmanager/src/EventManager.php:490}() /vagrant/src/cms/vendor/zendframework/zend-eventmanager/src/EventManager.php:490
    0.1441    3622800   7. RcmUser\Module->onBootstrap() /vagrant/src/cms/vendor/zendframework/zend-eventmanager/src/EventManager.php:490
    0.1441    3622848   8. Zend\ServiceManager\ServiceManager->get() /vagrant/src/cms/vendor/rcm/user/src/Module.php:48
    0.1441    3623824   9. Zend\ServiceManager\ServiceManager->create() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:532
    0.1441    3623824  10. Zend\ServiceManager\ServiceManager->doCreate() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:599
    0.1441    3623872  11. Zend\ServiceManager\ServiceManager->createFromFactory() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:636
    0.1446    3625984  12. Zend\ServiceManager\ServiceManager->createServiceViaCallback() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:1094
    0.1446    3626344  13. call_user_func:{/vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936}() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936
    0.1446    3626448  14. RcmUser\Service\Factory\EventListeners->createService() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936
    0.1446    3626632  15. Zend\ServiceManager\ServiceManager->get() /vagrant/src/cms/vendor/rcm/user/src/Service/Factory/EventListeners.php:54
    0.1446    3627576  16. Zend\ServiceManager\ServiceManager->create() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:532
    0.1446    3627576  17. Zend\ServiceManager\ServiceManager->doCreate() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:599
    0.1446    3627624  18. Zend\ServiceManager\ServiceManager->createFromFactory() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:636
    0.1457    3629784  19. Zend\ServiceManager\ServiceManager->createServiceViaCallback() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:1094
    0.1457    3630176  20. call_user_func:{/vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936}() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936
    0.1457    3630272  21. RcmUser\User\Service\Factory\UserDataServiceListeners->createService() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936
    0.1457    3630320  22. Zend\ServiceManager\ServiceManager->get() /vagrant/src/cms/vendor/rcm/user/src/User/Service/Factory/UserDataServiceListeners.php:50
    0.1457    3631240  23. Zend\ServiceManager\ServiceManager->create() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:532
    0.1457    3631240  24. Zend\ServiceManager\ServiceManager->doCreate() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:599
    0.1457    3631288  25. Zend\ServiceManager\ServiceManager->createFromFactory() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:636
    0.1463    3633440  26. Zend\ServiceManager\ServiceManager->createServiceViaCallback() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:1094
    0.1463    3633816  27. call_user_func:{/vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936}() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936
    0.1463    3633992  28. RcmUser\User\Service\Factory\DoctrineUserDataMapper->createService() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936
    0.1463    3634040  29. Zend\ServiceManager\ServiceManager->get() /vagrant/src/cms/vendor/rcm/user/src/User/Service/Factory/DoctrineUserDataMapper.php:49
    0.1463    3634960  30. Zend\ServiceManager\ServiceManager->create() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:532
    0.1463    3634960  31. Zend\ServiceManager\ServiceManager->doCreate() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:599
    0.1463    3635008  32. Zend\ServiceManager\ServiceManager->createFromFactory() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:636
    0.1468    3637048  33. Zend\ServiceManager\ServiceManager->createServiceViaCallback() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:1094
    0.1468    3637424  34. call_user_func:{/vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936}() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936
    0.1468    3637496  35. DoctrineORMModule\Service\EntityManagerAliasCompatFactory->createService() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936
    0.1468    3637544  36. Zend\ServiceManager\ServiceManager->get() /vagrant/src/cms/vendor/doctrine/doctrine-orm-module/src/DoctrineORMModule/Service/EntityManagerAliasCompatFactory.php:44
    0.1468    3638632  37. Zend\ServiceManager\ServiceManager->create() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:532
    0.1469    3638632  38. Zend\ServiceManager\ServiceManager->doCreate() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:599
    0.1469    3638816  39. Zend\ServiceManager\ServiceManager->createFromAbstractFactory() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:644
    0.1469    3639864  40. Zend\ServiceManager\ServiceManager->createServiceViaCallback() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:1126
    0.1469    3640256  41. call_user_func:{/vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936}() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936
    0.1469    3640352  42. DoctrineModule\ServiceFactory\AbstractDoctrineServiceFactory->createServiceWithName() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936
    0.1479    3649328  43. DoctrineORMModule\Service\EntityManagerFactory->createService() /vagrant/src/cms/vendor/doctrine/doctrine-module/src/DoctrineModule/ServiceFactory/AbstractDoctrineServiceFactory.php:59
    0.1487    3657552  44. Zend\ServiceManager\ServiceManager->get() /vagrant/src/cms/vendor/doctrine/doctrine-orm-module/src/DoctrineORMModule/Service/EntityManagerFactory.php:36
    0.1487    3658640  45. Zend\ServiceManager\ServiceManager->create() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:532
    0.1487    3658640  46. Zend\ServiceManager\ServiceManager->doCreate() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:599
    0.1487    3658688  47. Zend\ServiceManager\ServiceManager->createFromAbstractFactory() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:644
    0.1487    3659520  48. Zend\ServiceManager\ServiceManager->createServiceViaCallback() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:1126
    0.1487    3659896  49. call_user_func:{/vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936}() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936
    0.1488    3659936  50. DoctrineModule\ServiceFactory\AbstractDoctrineServiceFactory->createServiceWithName() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936
    0.1495    3665072  51. DoctrineORMModule\Service\DBALConnectionFactory->createService() /vagrant/src/cms/vendor/doctrine/doctrine-module/src/DoctrineModule/ServiceFactory/AbstractDoctrineServiceFactory.php:59
    0.1500    3679936  52. Zend\ServiceManager\ServiceManager->get() /vagrant/src/cms/vendor/doctrine/doctrine-orm-module/src/DoctrineORMModule/Service/DBALConnectionFactory.php:57
    0.1500    3681064  53. Zend\ServiceManager\ServiceManager->create() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:532
    0.1500    3681064  54. Zend\ServiceManager\ServiceManager->doCreate() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:599
    0.1500    3681112  55. Zend\ServiceManager\ServiceManager->createFromAbstractFactory() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:644
    0.1500    3681944  56. Zend\ServiceManager\ServiceManager->createServiceViaCallback() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:1126
    0.1500    3682336  57. call_user_func:{/vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936}() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936
    0.1500    3682376  58. DoctrineModule\ServiceFactory\AbstractDoctrineServiceFactory->createServiceWithName() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936
    0.1509    3690608  59. DoctrineORMModule\Service\ConfigurationFactory->createService() /vagrant/src/cms/vendor/doctrine/doctrine-module/src/DoctrineModule/ServiceFactory/AbstractDoctrineServiceFactory.php:59
    0.1531    3772664  60. Zend\ServiceManager\ServiceManager->get() /vagrant/src/cms/vendor/doctrine/doctrine-orm-module/src/DoctrineORMModule/Service/ConfigurationFactory.php:67
    0.1532    3775632  61. Zend\ServiceManager\ServiceManager->create() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:532
    0.1532    3775632  62. Zend\ServiceManager\ServiceManager->doCreate() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:599
    0.1532    3775680  63. Zend\ServiceManager\ServiceManager->createFromFactory() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:636
    0.1536    3777808  64. Zend\ServiceManager\ServiceManager->createServiceViaCallback() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:1094
    0.1536    3778184  65. call_user_func:{/vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936}() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936
    0.1536    3778264  66. Rcm\Factory\DoctrineCacheFactory->createService() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936
    0.1536    3778312  67. Zend\ServiceManager\ServiceManager->get() /vagrant/src/cms/vendor/rcm/core/src/Factory/DoctrineCacheFactory.php:52
    0.1536    3779248  68. Zend\ServiceManager\ServiceManager->create() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:532
    0.1536    3779248  69. Zend\ServiceManager\ServiceManager->doCreate() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:599
    0.1536    3779296  70. Zend\ServiceManager\ServiceManager->createFromFactory() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:636
    0.1540    3781368  71. Zend\ServiceManager\ServiceManager->createServiceViaCallback() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:1094
    0.1540    3781792  72. call_user_func:{/vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936}() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936
    0.1540    3781880  73. Rcm\Factory\CacheFactory->createService() /vagrant/src/cms/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936
    0.1543    3787688  74. Zend\Cache\StorageFactory::factory() /vagrant/src/cms/vendor/rcm/core/src/Factory/CacheFactory.php:62
    0.1543    3787808  75. Zend\Cache\StorageFactory::adapterFactory() /vagrant/src/cms/vendor/zendframework/zend-cache/src/StorageFactory.php:72
    0.1578    3906192  76. Zend\Cache\Storage\Adapter\Memcached->setOptions() /vagrant/src/cms/vendor/zendframework/zend-cache/src/StorageFactory.php:149
    0.1596    3946352  77. Zend\Cache\Storage\Adapter\AbstractAdapter->setOptions() /vagrant/src/cms/vendor/zendframework/zend-cache/src/Storage/Adapter/Memcached.php:131
    0.1604    3956008  78. Zend\EventManager\EventManager->triggerEvent() /vagrant/src/cms/vendor/zendframework/zend-cache/src/Storage/Adapter/AbstractAdapter.php:130
    0.1604    3956040  79. Zend\EventManager\EventManager->triggerListeners() /vagrant/src/cms/vendor/zendframework/zend-eventmanager/src/EventManager.php:251
    0.1604    3956416  80. Zend\EventManager\EventManager->getSharedListeners() /vagrant/src/cms/vendor/zendframework/zend-eventmanager/src/EventManager.php:469
    0.1604    3956992  81. Zend\EventManager\SharedEventManager->getListeners() /vagrant/src/cms/vendor/zendframework/zend-eventmanager/src/EventManager.php:530
    0.1604    3957040  82. array_key_exists() /vagrant/src/cms/vendor/zendframework/zend-eventmanager/src/SharedEventManager.php:153
```

**Comment by wshafer** [comment]
@Ocramius Yea I agree it's a huge jump.  Not quite the dot release here.

**Comment by Ocramius** [comment]
Semver should be ok with that. Anyway, problem doesn't come from the eventmanager diffs: https://github.com/zendframework/zend-eventmanager/compare/release-2.5.2...release-2.6.2

**Comment by wshafer** [comment]
The problems probably not coming from ZF at all really.  The problem is most likely a BC break and I need to update one of my listeners.  But as you can see... I'm not sure how to track that one down.

**Comment by Ocramius** [comment]
All these upgrades are within BC boundaries: this is 100% a bug introduced in zf

**Comment by wshafer** [comment]
This is interesting... One of the listeners IS the Zend\EventManager\SharedEventManager

**Comment by wshafer** [comment]
Here's the culprit!   
vendor/zendframework/zend-cache/src/Storage/Adapter/AbstractAdapter.php:190

```php
public function getEventManager()
    {
        if ($this->events === null) {
            $this->events = new EventManager(new SharedEventManager(), [__CLASS__, get_class($this)]);
        }
        return $this->events;
    }
```

**Comment by pensiero** [comment]
Same here, one of the listeners is `Zend\ModuleManager\ModuleManager`

Full stack trace:
```
PHP Warning array_key_exists(): The first argument should be either a string or an integer 
    [internal] array_key_exists
    /var/www/vendor/zendframework/zend-eventmanager/src/SharedEventManager.php:153 Zend\EventManager\SharedEventManager::getListeners
    /var/www/vendor/zendframework/zend-eventmanager/src/EventManager.php:530 Zend\EventManager\EventManager::getSharedListeners
    /var/www/vendor/zendframework/zend-eventmanager/src/EventManager.php:469 Zend\EventManager\EventManager::triggerListeners
    /var/www/vendor/zendframework/zend-eventmanager/src/EventManager.php:251 Zend\EventManager\EventManager::triggerEvent
    /var/www/vendor/zendframework/zend-cache/src/Storage/Adapter/AbstractAdapter.php:130 Zend\Cache\Storage\Adapter\AbstractAdapter::setOptions
    /var/www/vendor/zendframework/zend-cache/src/Storage/Adapter/Redis.php:119 Zend\Cache\Storage\Adapter\Redis::setOptions
    /var/www/vendor/zendframework/zend-cache/src/Storage/Adapter/AbstractAdapter.php:80 Zend\Cache\Storage\Adapter\AbstractAdapter::__construct
    /var/www/vendor/zendframework/zend-cache/src/Storage/Adapter/Redis.php:67 Zend\Cache\Storage\Adapter\Redis::__construct
    /var/www/module/Uala/src/Uala/Cache/RedisFactory.php:40 Uala\Cache\RedisFactory::createService
    [internal] call_user_func
    /var/www/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936 Zend\ServiceManager\ServiceManager::createServiceViaCallback
    /var/www/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:1094 Zend\ServiceManager\ServiceManager::createFromFactory
    /var/www/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:636 Zend\ServiceManager\ServiceManager::doCreate
    /var/www/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:599 Zend\ServiceManager\ServiceManager::create
    /var/www/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:532 Zend\ServiceManager\ServiceManager::get
    /var/www/module/Uala/src/Uala/ServiceFactory/ArtaxServiceFactory.php:19 Uala\ServiceFactory\ArtaxServiceFactory::createService
    [internal] call_user_func
    /var/www/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936 Zend\ServiceManager\ServiceManager::createServiceViaCallback
    /var/www/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:1094 Zend\ServiceManager\ServiceManager::createFromFactory
    /var/www/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:636 Zend\ServiceManager\ServiceManager::doCreate
    /var/www/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:599 Zend\ServiceManager\ServiceManager::create
    /var/www/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:532 Zend\ServiceManager\ServiceManager::get
    /var/www/module/Uala/src/Uala/ServiceFactory/VenueServiceFactory.php:19 Uala\ServiceFactory\VenueServiceFactory::createService
    [internal] call_user_func
    /var/www/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936 Zend\ServiceManager\ServiceManager::createServiceViaCallback
    /var/www/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:1094 Zend\ServiceManager\ServiceManager::createFromFactory
    /var/www/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:636 Zend\ServiceManager\ServiceManager::doCreate
    /var/www/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:599 Zend\ServiceManager\ServiceManager::create
    /var/www/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:532 Zend\ServiceManager\ServiceManager::get
    /var/www/module/Uala/src/Uala/Service/EntityService.php:27 Uala\Service\EntityService::__construct
    /var/www/module/Uala/src/Uala/RouteFactory/EntityRouteFactory.php:19 Uala\RouteFactory\EntityRouteFactory::createService
    [internal] call_user_func
    /var/www/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:936 Zend\ServiceManager\ServiceManager::createServiceViaCallback
    /var/www/vendor/zendframework/zend-servicemanager/src/AbstractPluginManager.php:330 Zend\ServiceManager\AbstractPluginManager::createServiceViaCallback
    /var/www/vendor/zendframework/zend-servicemanager/src/AbstractPluginManager.php:287 Zend\ServiceManager\AbstractPluginManager::createFromFactory
    /var/www/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:636 Zend\ServiceManager\ServiceManager::doCreate
    /var/www/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:599 Zend\ServiceManager\ServiceManager::create
    /var/www/vendor/zendframework/zend-servicemanager/src/ServiceManager.php:532 Zend\ServiceManager\ServiceManager::get
    /var/www/vendor/zendframework/zend-servicemanager/src/AbstractPluginManager.php:161 Zend\ServiceManager\AbstractPluginManager::get
    /var/www/vendor/zendframework/zend-mvc/src/Router/SimpleRouteStack.php:274 Zend\Mvc\Router\SimpleRouteStack::routeFromArray
    /var/www/vendor/zendframework/zend-mvc/src/Router/Http/TreeRouteStack.php:163 Zend\Mvc\Router\Http\TreeRouteStack::routeFromArray
    /var/www/vendor/zendframework/zend-mvc/src/Router/Http/TreeRouteStack.php:113 Zend\Mvc\Router\Http\TreeRouteStack::addRoute
    /var/www/vendor/zendframework/zend-mvc/src/Router/SimpleRouteStack.php:140 Zend\Mvc\Router\SimpleRouteStack::addRoutes
    /var/www/vendor/zendframework/zend-mvc/src/Router/Http/Part.php:143 Zend\Mvc\Router\Http\Part::match
    /var/www/vendor/zendframework/zend-mvc/src/Router/Http/TreeRouteStack.php:279 Zend\Mvc\Router\Http\TreeRouteStack::match
    /var/www/vendor/zendframework/zend-mvc/src/RouteListener.php:44 Zend\Mvc\RouteListener::onRoute
    [internal] call_user_func
    /var/www/vendor/zendframework/zend-eventmanager/src/EventManager.php:490 Zend\EventManager\EventManager::triggerListeners
    /var/www/vendor/zendframework/zend-eventmanager/src/EventManager.php:214 Zend\EventManager\EventManager::trigger
    /var/www/vendor/zendframework/zend-mvc/src/Application.php:297 Zend\Mvc\Application::run
    /var/www/public/index.php:23 [main]
```

**Comment by weierophinney** [comment]
This is a duplicate of zendframework/zend-cache#71, which is (hopefully) resolved with zendframework/zend-cache#73. (The problem is with lazy-loading an EventManager instance within zend-cache; there was no coverage of that particular call, which meant that an errant v3 instantiation pattern was missed when we tagged zend-cache 2.6.0.)

If you could, please test the patch in that latter link, and let me know if it resolves your issue.

**Comment by wshafer** [comment]
@weierophinney - Update to Zend-Cache 2.6.1 does resolve this for me.  Thanks for the quick fix.
